core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1493,6 +1515,33 @@ private void unparseExtractFunction(SqlWriter writer, SqlCall call, int leftPrec
     }
   }
 
+  private boolean isDateTimeCast(SqlNode operand) {
+    boolean isCastCall = ((SqlBasicCall) operand).getOperator() == CAST;
+    boolean isDateTimeCast = isCastCall
+        && ((SqlDataTypeSpec) ((SqlBasicCall) operand).operands[1])
+            .getTypeName().toString().equals("TIMESTAMP");
```

Comment:
Can we use `SqlTypeName.TIMESTAMP` for comparison?

Reply:
No here it do not gives SqlTypeName